### PR TITLE
fix(build):change npm ci to yarn

### LINKED ISSUE #8715: 最新仓库的 b5be05 自构建 docker image 失败： The `npm ci` command can only install with an existing package-lock.json
### 路由地址

/rsshub/routes

### 完整路由地址

/rsshub/routes

### 相关文档

https://docs.rsshub.app/install/

### 预期是什么？

尝试使用 `docker compose up -d --build ` 编译镜像并运行。

### 实际发生了什么？

```bash
➜  rsshub docker-compose up -d --build
Sending build context to Docker daemon  1.306MB
Step 1/24 : FROM node:14-buster-slim as dep-builder
 ---> eb79fdb26bfb
Step 2/24 : LABEL MAINTAINER https://github.com/DIYgod/RSSHub/
 ---> Using cache
 ---> c4749b04b7de
Step 3/24 : ARG USE_CHINA_NPM_REGISTRY=0;
 ---> Using cache
 ---> c1c517d40271
Step 4/24 : ARG PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=1;
 ---> Using cache
 ---> 444118db784f
Step 5/24 : RUN ln -sf /bin/bash /bin/sh
 ---> Using cache
 ---> 14503b0f1f74
Step 6/24 : RUN apt-get update && apt-get install -yq libgconf-2-4 apt-transport-https git dumb-init python3 build-essential --no-install-recommends
 ---> Using cache
 ---> 2fdbbbe8f179
Step 7/24 : WORKDIR /app
 ---> Using cache
 ---> 54adccefa424
Step 8/24 : COPY . /app
 ---> 16bf21ba9903
Step 9/24 : RUN if [ "$USE_CHINA_NPM_REGISTRY" = 1 ]; then   echo 'use npm mirror'; npm config set registry https://registry.npm.taobao.org;   fi;
 ---> Running in 50c555765df0
 ---> 1bca805ecd7c
Step 10/24 : RUN npm i -g npm
 ---> Running in 01837ca3d36d
/usr/local/bin/npm -> /usr/local/lib/node_modules/npm/bin/npm-cli.js
/usr/local/bin/npx -> /usr/local/lib/node_modules/npm/bin/npx-cli.js
+ npm@8.3.0
added 67 packages from 17 contributors, removed 290 packages and updated 147 packages in 11.3s
 ---> 9d89a70cda08
Step 11/24 : RUN if [ "$PUPPETEER_SKIP_CHROMIUM_DOWNLOAD" = 0 ]; then   unset PUPPETEER_SKIP_CHROMIUM_DOWNLOAD && npm ci ;  else   export PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true && npm ci ;  fi;
 ---> Running in 25d48fc7f353
npm ERR! The `npm ci` command can only install with an existing package-lock.json or
npm ERR! npm-shrinkwrap.json with lockfileVersion >= 1. Run an install with npm@5 or
npm ERR! later to generate a package-lock.json file, then try again.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-12-12T07_51_56_694Z-debug-0.log
1 error occurred:
        * Status: The command '/bin/sh -c if [ "$PUPPETEER_SKIP_CHROMIUM_DOWNLOAD" = 0 ]; then   unset PUPPETEER_SKIP_CHROMIUM_DOWNLOAD && npm ci ;  else   export PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true && npm ci ;  fi;' returned a non-zero code: 1, Code: 1
```

似乎是在 Dockerfile 的这一步发生了错误

```
RUN if [ "$PUPPETEER_SKIP_CHROMIUM_DOWNLOAD" = 0 ]; then \
  unset PUPPETEER_SKIP_CHROMIUM_DOWNLOAD && npm ci ;\
  else \
  export PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true && npm ci ;\
  fi;
```

### 部署

自建

### 部署相关信息

Docker version 20.10.11, build dea9396 Docker Compose version v2.1.1

### 额外信息

```shell
docker-compose.yml

version: '3'
services:
  rsshub:
    build: ./RSSHub
    container_name: rsshub
    restart: unless-stopped
    ports:
      - 1200:1200
    environment:
      - NODE_ENV=production
      - CACHE_TYPE=null
      - PUPPETEER_WS_ENDPOINT=ws://browserless:3000
      - DUBUG_INFO=false
      - TIME_ZONE=Asia/Shanghai
    env_file:
      - ./rsshub.env
    depends_on:
      - browserless

  browserless:
    image: browserless/chrome
    restart: unless-stopped
    ports:
      - "30001:3000"

networks:
  default:
    external: true
    name: net


```

```
➜  rsshub tree -L 1 ./
./
|-- RSSHub
|-- docker-compose.yml
`-- rsshub.env

1 directory, 2 files
```

### 这不是重复的 issue

- [X] 我已经搜索了[现有 issue](https://github.com/DIYgod/RSSHub/issues)，以确保该错误尚未被报告。

### PR DESCRIPTION
**Pull request by RikkaBlue**
## 该 PR 相关 Issue / Involved issue

Close #8715 

## 完整路由地址 / Example for the proposed route(s)

```routes
NOROUTE
```

## 新RSS检查列表 / New RSS Script Checklist
  
- [ ] New Route
- [ ] Documentation
  - [ ] CN
  - [ ] EN
- [ ] 全文获取 fulltext
  - [ ] Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] 日期和时间 date and time
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added 
- [ ] `Puppeteer`

## 说明 / Note
顺便改了一下 Dockerfile 的命令顺序，在仅更改路由而不改动 package.json 的情况下，利用缓存 Docker 层的优势[^1]

[^1]:[Building Efficient Dockerfiles - Node.js](http://bitjudo.com/blog/2014/03/13/building-efficient-dockerfiles-node-dot-js/)

### CODE REVIEW

**Review by SukkaW — APPROVED** [review]
LGTM.

File: Dockerfile [hunk 1]
```diff
@@ -10,7 +10,11 @@ RUN ln -sf /bin/bash /bin/sh
 RUN apt-get update && apt-get install -yq libgconf-2-4 apt-transport-https git dumb-init python3 build-essential --no-install-recommends
 
 WORKDIR /app
-COPY . /app
+
+COPY ./yarn.lock /app
+COPY ./package.json /app
+
+#COPY . /app
```
Comment (TonyRL) [inline Dockerfile:17]:
```suggestion

```
Remove unused command

**Comment by RikkaBlue** [comment]
@TonyRL  意外关闭 PR 了，麻烦重新开启一下，谢谢。

**Comment by NeverBehave** [comment]
@SukkaW 所以又为啥换成了yarn

**Comment by AgFlore** [comment]
> @SukkaW 所以又为啥换成了yarn

大概要问 @DIYgod 
https://github.com/DIYgod/RSSHub/commit/483ae4593bc9aea66259035f225db73258a32720
https://github.com/DIYgod/RSSHub/commit/6e10cb59ba27e9941cc0997c5e3dc528fb3ac02f

**Comment by SukkaW** [comment]
> @SukkaW 所以又为啥换成了yarn

问我干嘛。换我，我肯定用 `pnpm-lock.yaml` 啊

**Comment by KTachibanaM** [comment]
Is there activity on this MR? My MR seems to be blocked on this one. Thanks.

**Pull request merged**